src/libraries/System.Reflection.Emit/tests/ModuleBuilder/ModuleBuilderCreateGlobalFunctions.cs

```diff
@@ -17,6 +17,7 @@ public void CreateGlobalFunctions_SingleGlobalMethod()
             ilGenerator.Emit(OpCodes.Ret);
 
             module.CreateGlobalFunctions();
+            Assert.Null(method.DeclaringType);
```

Comment:
The asserts added to this file fails on mono, https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-72180-merge-437d30d8adc24f8fb1/System.Reflection.Emit.Tests/1/console.5d48a2e1.log?helixlogtype=result

Mono implementation looks quite different, seems its better revert these asserts, the same happening 
for Asserts in `ModuleBuilderDefineInitializedData.cs`

Reply:
Looks like Mono bugs. An issue should be opened on this and the asserts can be kept under `if (!PlatformDetection.IsMonoRuntime) // [ActiveIssue("https://github.com/dotnet/runtime/issues/xxxxxx")]`

Reply:
Issue opened, and Failing tests on Mono annotated. https://github.com/dotnet/runtime/issues/72858

Reply:
@JosieBigler we want the tests keep running on mono except the new Assert, please move only the `Assert.Null(method.DeclaringType);` part into `if (!PlatformDetection.IsMonoRuntime)` conditional:
```suggestion
        if (!PlatformDetection.IsMonoRuntime) // [ActiveIssue("https://github.com/dotnet/runtime/issues/xxxxxx")]
        {
            Assert.Null(method.DeclaringType);
        }
```